core/frontend/src/test/tile/ImdlParser.test.ts

```diff
@@ -278,4 +278,175 @@ describe("ImdlParser", () => {
+      0x30, 0x34, 0x32, 0x36, 0x33, 0x2C, 0x2D, 0x30, 0x2E, 0x30, 0x35, 0x30,
+      0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x34,
+      0x32, 0x36, 0x33, 0x2C, 0x2D, 0x33, 0x32, 0x2E, 0x31, 0x30, 0x30, 0x30,
+      0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x31, 0x5D,
+      0x7D, 0x2C, 0x22, 0x75, 0x6E, 0x69, 0x66, 0x6F, 0x72, 0x6D, 0x43, 0x6F,
+      0x6C, 0x6F, 0x72, 0x22, 0x3A, 0x31, 0x36, 0x37, 0x37, 0x37, 0x32, 0x31,
+      0x35, 0x2C, 0x22, 0x77, 0x69, 0x64, 0x74, 0x68, 0x22, 0x3A, 0x38, 0x34,
+      0x7D, 0x7D, 0x5D, 0x7D, 0x7D, 0x2C, 0x22, 0x6E, 0x6F, 0x64, 0x65, 0x73,
+      0x22, 0x3A, 0x7B, 0x22, 0x4E, 0x6F, 0x64, 0x65, 0x5F, 0x52, 0x6F, 0x6F,
+      0x74, 0x22, 0x3A, 0x22, 0x4D, 0x65, 0x73, 0x68, 0x5F, 0x52, 0x6F, 0x6F,
+      0x74, 0x22, 0x7D, 0x2C, 0x22, 0x73, 0x63, 0x65, 0x6E, 0x65, 0x22, 0x3A,
+      0x22, 0x64, 0x65, 0x66, 0x61, 0x75, 0x6C, 0x74, 0x53, 0x63, 0x65, 0x6E,
+      0x65, 0x22, 0x2C, 0x22, 0x73, 0x63, 0x65, 0x6E, 0x65, 0x73, 0x22, 0x3A,
+      0x7B, 0x22, 0x64, 0x65, 0x66, 0x61, 0x75, 0x6C, 0x74, 0x53, 0x63, 0x65,
+      0x6E, 0x65, 0x22, 0x3A, 0x7B, 0x22, 0x6E, 0x6F, 0x64, 0x65, 0x73, 0x22,
+      0x3A, 0x5B, 0x22, 0x72, 0x6F, 0x6F, 0x74, 0x4E, 0x6F, 0x64, 0x65, 0x22,
+      0x5D, 0x7D, 0x7D, 0x7D, 0x0A, 0x20, 0x20, 0x20, 0xA0, 0x05, 0x00, 0x3F,
+      0x3F, 0x33, 0x17, 0x18, 0x17, 0x18, 0x08, 0x07, 0x17, 0x85, 0xCC, 0xC0,
+      0x00, 0x00, 0x86, 0x18, 0x17, 0x05, 0x00, 0x3F, 0x3F, 0x30, 0xC2, 0xC1,
+      0xC2, 0xC1, 0xC2, 0xC1, 0xC2, 0x0C, 0xC0, 0x00, 0x00, 0xC1, 0xC2, 0x06,
+      0x08, 0x07, 0x87, 0x00, 0x00, 0x87, 0x80, 0x78, 0xF3, 0x00, 0x00, 0x00,
+      0x41, 0x42, 0x07, 0x05, 0x33, 0xF0, 0x0F, 0xCF, 0x8A, 0x89, 0x8A, 0x89,
+      0x8A, 0x89, 0x8A, 0x89, 0x8A, 0x03, 0x00, 0x00, 0x00, 0x89, 0x01, 0x3C,
+      0xFC, 0xCC, 0x03, 0x37, 0x31, 0x6A, 0x2F, 0x39, 0x6A, 0x69, 0x6A, 0x02,
+      0x07, 0x50, 0xE7, 0x50, 0xE0, 0xD0, 0x00, 0x0E, 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x01, 0x00, 0x30, 0x0C, 0x03, 0xFD, 0xFE, 0xFD, 0x01, 0x00,
+      0x30, 0x00, 0x03, 0xFE, 0xFD, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0C,
+      0x9F, 0xFC, 0xBB, 0x01, 0x52, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xFF,
+      0x80, 0x00, 0x00, 0x00, 0xD1, 0x00, 0x04, 0x04, 0x06, 0x08, 0x04, 0x06,
+      0x02, 0x10, 0x04, 0x04, 0x04, 0x06, 0x0C, 0x04, 0x0E, 0x10, 0x0A, 0x10,
+      0x04, 0x04, 0x04, 0x04, 0x0A, 0x08, 0x06, 0x02, 0x14, 0x04, 0x04, 0x06,
+      0x0C, 0x04, 0x0E, 0x10, 0x04, 0x12, 0x14, 0x0E, 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x8A, 0x20, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+    ]);
+
+    const parser = acquireImdlParser({});
+    const document = await parser.parse({
+      data: tileData,
+      batchModelId: "0x123",
+      is3d: true,
+      maxVertexTableSize: 4096,
+    });
+
+    expect(typeof document).toEqual("object");
```

Comment:
You don't want to verify you correctly decoded the buffer(s)?

Reply:
I could but I'm using this code in image tests already with compressed tiles, so I did not bother

Reply:
Is it difficult to verify here? If somebody breaks it and you're relying solely on image tests to tell you, you won't find out until PR merges, nightly run fails, somebody investigates and tracks it down to a meshopt problem and finds the offending PR.

Reply:
will add a test in a following PR